components/datagrid/HeaderRow.js

```diff
@@ -0,0 +1,40 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Row from './Row';
+import HeaderCell from './HeaderCell';
+import Checkbox from '../checkbox';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class HeaderRow extends PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.any,
+    onSelectionChange: PropTypes.func,
+    selectable: PropTypes.bool,
+    selected: PropTypes.bool,
+    sliceFrom: PropTypes.number,
+    sliceTo: PropTypes.number,
+  };
+
+  render() {
+    const { className, children, sliceFrom, sliceTo, onSelectionChange, selected, selectable, ...others } = this.props;
+
+    const childrenArray = Array.isArray(children) ? children : [children];
```

Comment:
Guess what 💯

Reply:
This doesn't work because `children` can be either an `object` or an `array`. The spread operator can not handle mixed types.

 :ok: by @driesd on 23/01/2018 15:35:28